test/api/dependency/api_test.go

```diff
@@ -101,3 +103,106 @@ func TestDependencyList(t *testing.T) {
 		assert.Must(t, Application.Delete(dependency.To.ID))
 	}
 }
+
+func TestReverseDependency(t *testing.T) {
+	for _, r := range ReverseSamples {
+		// Create Applications.
+		assert.Must(t, Application.Create(&r.Application1))
+		assert.Must(t, Application.Create(&r.Application2))
+		assert.Must(t, Application.Create(&r.Application3))
+
+		firstDependency := api.Dependency{
+			From: api.Ref{
+				ID: r.Application1.ID,
+			},
+			To: api.Ref{
+				ID: r.Application2.ID,
+			},
+		}
+		err := Dependency.Create(&firstDependency)
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+
+		secondDependency := api.Dependency{
+			From: api.Ref{
+				ID: r.Application2.ID,
+			},
+			To: api.Ref{
+				ID: r.Application3.ID,
+			},
+		}
+		err = Dependency.Create(&secondDependency)
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+
+		// Indirect Reverse dependency should fail.
+		thirdDependency := api.Dependency{
+			From: api.Ref{
+				ID: r.Application3.ID,
+			},
+			To: api.Ref{
+				ID: r.Application1.ID,
+			},
+		}
+		err = Dependency.Create(&thirdDependency)
+		if err == nil {
+			t.Error("Indirect Reverse dependency not allowed")
+		}
+
+		fourthDependency := api.Dependency{
+			From: api.Ref{
+				ID: r.Application1.ID,
+			},
+			To: api.Ref{
+				ID: r.Application3.ID,
+			},
+		}
+		err = Dependency.Create(&fourthDependency)
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+
+		// Direct Reverse dependency should fail.
+		fifthDependency := api.Dependency{
+			From: api.Ref{
+				ID: r.Application2.ID,
+			},
+			To: api.Ref{
+				ID: r.Application1.ID,
+			},
+		}
+		err = Dependency.Create(&fifthDependency)
+		if err == nil {
+			t.Error("Direct Reverse dependency not allowed")
+		}
+
+		// Direct Reverse dependency should fail.
+		sixthDependency := api.Dependency{
+			From: api.Ref{
+				ID: r.Application3.ID,
+			},
+			To: api.Ref{
+				ID: r.Application2.ID,
+			},
+		}
+		err = Dependency.Create(&sixthDependency)
+		if err == nil {
+			t.Error("Direct Reverse dependency not allowed")
+		}
+
+		// Delete Dependencies.
+		assert.Should(t, Dependency.Delete(firstDependency.ID))
+		assert.Should(t, Dependency.Delete(secondDependency.ID))
+		assert.Should(t, Dependency.Delete(thirdDependency.ID))
```

Comment:
Even API works in the way, that DELETE on not existing resources doesn't fail, just do nothing, it might be cleaner not delete dependencies that were not created.

Reply:
Okay will remove the lines